packages/web-client/app/services/card-customization-options.ts

```diff
@@ -0,0 +1,123 @@
+import Service from '@ember/service';
+import config from '../config/environment';
+import {
+  ColorCustomizationOption,
+  PatternCustomizationOption,
+} from '../utils/web3-strategies/types';
+import { task } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import { tracked } from '@glimmer/tracking';
+
+interface ColorCustomizationResponseObject {
+  attributes: {
+    'pattern-color': string;
+    'text-color': string;
+    background: string;
+    description?: string;
+  };
+  id: string;
+}
+
+interface PatternCustomizationResponseObject {
+  attributes: {
+    'pattern-url': string;
+    description?: string;
+  };
+  id: string;
+}
+
+// TODO: move this to orbit.js/ember-data
+let convertToColorCustomizationOption = (
+  o: ColorCustomizationResponseObject
+): ColorCustomizationOption => {
+  return {
+    patternColor: o.attributes['pattern-color'],
+    textColor: o.attributes['text-color'],
+    headerBackground: o.attributes.background,
+    description: o.attributes.description,
+    id: o.id,
+  };
+};
+
+let convertToPatternCustomizationOption = (
+  o: PatternCustomizationResponseObject
+): PatternCustomizationOption => {
+  return {
+    patternUrl: o.attributes['pattern-url'],
+    description: o.attributes.description,
+    id: o.id,
+  };
+};
+
+export default class CardCustomizationOptions extends Service {
+  @tracked loaded = false;
+  @tracked patternOptions: PatternCustomizationOption[] | null = [];
+  @tracked colorOptions: ColorCustomizationOption[] | null = [];
+
+  async ensureCustomizationOptionsLoaded() {
+    if (!this.loaded) {
+      return taskFor(this.fetchCustomizationOptions).perform();
+    }
+  }
+
+  @task *fetchCustomizationOptions(): any {
+    let colorOptions: ColorCustomizationOption[] = [];
+    let patternOptions: PatternCustomizationOption[] = [];
+    try {
+      let [_colorOptions, _patternOptions] = yield Promise.all([
+        fetch(`${config.hubURL}/api/prepaid-card-color-schemes`, {
+          method: 'GET',
+          headers: {
+            Accept: 'application/vnd.api+json',
+          },
+        })
+          .then((v) => v.json())
+          .then((v) => v.data.map(convertToColorCustomizationOption)),
```

Comment:
Extract fetch and response processing to a separate e-concurrency task so that async promise resolution is managed.